deltacat/tests/compactor/testcases.py

```diff
@@ -0,0 +1,307 @@
+            {
+                "key_name": "sk_col_1",
+            },
+        ],
+        [],
+        ["pk_col_1", "sk_col_1"],
+        [
+            pa.array([i for i in range(0, 10)]),
+            pa.array([str(i) for i in range(10)]),
+        ],
+        None,
+        ["1"],
+        pa.Table.from_arrays(
+            [
+                pa.array([i for i in range(0, 10)]),
+                pa.array([str(i) for i in range(10)]),
+            ],
+            names=["pk_col_1", "sk_col_1"],
+        ),
+        None,
+        None,
+        True,
+        False,
+        True,
+        MAX_RECORDS_PER_FILE,
+        None,
+    ),
+    "8-incremental-timestamp-pk-simple": (
+        BASE_TEST_SOURCE_TABLE_VERSION,
+        BASE_TEST_DESTINATION_TABLE_VERSION,
+        ["pk_col_1"],
+        [
+            {
+                "key_name": "sk_col_1",
+            },
+        ],
+        [],
+        ["pk_col_1", "sk_col_1"],
+        [
+            pa.array(offer_iso8601_timestamp_list(10, "minutes")),
+            pa.array([str(i) for i in range(10)]),
+        ],
+        None,
+        ["1"],
+        pa.Table.from_arrays(
+            [
+                pa.array(offer_iso8601_timestamp_list(10, "minutes")),
+                pa.array([str(i) for i in range(10)]),
+            ],
+            names=["pk_col_1", "sk_col_1"],
+        ),
+        None,
+        None,
+        True,
+        False,
+        True,
+        MAX_RECORDS_PER_FILE,
+        None,
+    ),
+    "9-incremental-decimal-timestamp-pk-multi": (
+        BASE_TEST_SOURCE_TABLE_VERSION,
+        BASE_TEST_DESTINATION_TABLE_VERSION,
+        ["pk_col_1", "pk_col_2"],
+        [
+            {
+                "key_name": "sk_col_1",
+            },
+        ],
+        [],
+        ["pk_col_1", "pk_col_2", "sk_col_1"],
+        [
+            pa.array([i / 10 for i in range(0, 20)]),
+            pa.array(offer_iso8601_timestamp_list(20, "minutes")),
+            pa.array([0.1] * 4 + [0.2] * 4 + [0.3] * 4 + [0.4] * 4 + [0.5] * 4),
+        ],
+        None,
+        ["1"],
+        pa.Table.from_arrays(
+            [
+                pa.array([i / 10 for i in range(0, 20)]),
+                pa.array(offer_iso8601_timestamp_list(20, "minutes")),
+                pa.array([0.1] * 4 + [0.2] * 4 + [0.3] * 4 + [0.4] * 4 + [0.5] * 4),
+            ],
+            names=["pk_col_1", "pk_col_2", "sk_col_1"],
+        ),
+        None,
+        None,
+        True,
+        False,
+        True,
+        MAX_RECORDS_PER_FILE,
+        None,
+    ),
+    "10-incremental-decimal-pk-multi-dup": (
+        BASE_TEST_SOURCE_TABLE_VERSION,
+        BASE_TEST_DESTINATION_TABLE_VERSION,
+        ["pk_col_1"],
+        [
+            {
+                "key_name": "sk_col_1",
+            },
+        ],
+        [],
+        ["pk_col_1", "sk_col_1"],
+        [
+            pa.array([0.1] * 4 + [0.2] * 4 + [0.3] * 4 + [0.4] * 4 + [0.5] * 4),
+            pa.array([i for i in range(20)]),
```

Comment:
Reverse the order so we know sorting works.

Reply:
Done